ui/perfherder/App.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { HashRouter, Route, Switch, Redirect } from 'react-router-dom';
+import { hot } from 'react-hot-loader/root';
+
+import GraphsView from './graphs/GraphsView';
+import { AlertsView } from './alerts/AlertsView';
+import { CompareView } from './compare/CompareView';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // keep track of the mainviews graph and table data so the API won't be
+    // called again when navigating back from bugdetailsview.
+    this.state = {
+      // graphData: null,
+      // tableData: null,
+    };
+  }
+
+  updateAppState = state => {
+    this.setState(state);
+  };
+
+  render() {
+    return (
+      <HashRouter>
+        <main>
+          <Switch>
+            <Route
+              exact
+              path="/alerts"
+              render={props => <AlertsView {...props} />}
+            />
+            <Route
+              path="/alerts?id=:id&status=:status&framework=:framework&filter=:filter&hideImprovements=:hideImprovements&hideDwnToInv=:hideDwnToInv&page=:page"
+              render={props => (
+                <AlertsView
+                  {...props}
+                  // mainGraphData={this.state.graphData}
+                  // mainTableData={this.state.tableData}
+                  // updateAppState={this.updateAppState}
+                />
+              )}
+            />
+            <Route path="/graphs" component={GraphsView} />
+            <Route
+              path="/graphs?timerange=:timerange&series=:series&highlightedRevisions=:highlightedRevisions&highlightAlerts=:highlightAlerts&zoom=:zoom&selected=:selected"
+              component={GraphsView}
+            />
+            <Route path="/compare" component={CompareView} />
+            {/* <Route
```

Comment:
Is this for something coming up in a future PR?

Reply:
oops, it is... so strange. Let me remove that stuff :) Clearly I'm ready for a vacation!